include/RevProc.h

```diff
@@ -696,6 +674,15 @@ class RevProc{
     return isFloat ? regFile->FP_Scoreboard.any() : regFile->RV_Scoreboard.any();
   }
 
+  bool ThreadHasDependencies(const uint32_t& ThreadID) const {
+    // Make sure this ThreadID is in AssignedThreads
+    if (AssignedThreads.find(ThreadID) == AssignedThreads.end()) {
+      output->fatal(CALL_INFO, -1, "Error: Tried to check if thread %" PRIu32 " has dependencies but that thread was not found in AssignedThreads. This is a bug\n", ThreadID);
+    }
+    const RevRegFile* regFile = AssignedThreads.at(ThreadID)->GetRegFile();
+    return regFile->FP_Scoreboard.any() || regFile->RV_Scoreboard.any();
+  }
```

Comment:
See above. I'd switch order of `FP` and `RV`. This might make the above function redundant; please remove it if it's not needed. It's just another untested possible failure point if it's not used.

Reply:
Yup above function is removed now. I switched the order in this one